packages/sources/por-address-list/src/transport/address.ts

```diff
@@ -0,0 +1,102 @@
+import { SubscriptionTransport } from '@chainlink/external-adapter-framework/transports/abstract/subscription'
+import { EndpointContext } from '@chainlink/external-adapter-framework/adapter'
+import { PoRAddress } from '@chainlink/external-adapter-framework/adapter/por'
+import { TransportDependencies } from '@chainlink/external-adapter-framework/transports'
+import { AdapterResponse, sleep } from '@chainlink/external-adapter-framework/util'
+import { POR_ADDRESS_LIST_ABI } from '../config/abi'
+import { BaseEndpointTypes, inputParameters } from '../endpoint/address'
+import { ethers } from 'ethers'
+import { fetchAddressList } from './utils'
+
+export type AddressTransportTypes = BaseEndpointTypes
+
+type RequestParams = typeof inputParameters.validated
+
+export class AddressTransport extends SubscriptionTransport<AddressTransportTypes> {
+  provider!: ethers.providers.JsonRpcProvider
+
+  async initialize(
+    dependencies: TransportDependencies<AddressTransportTypes>,
+    adapterSettings: AddressTransportTypes['Settings'],
+    endpointName: string,
+    transportName: string,
+  ): Promise<void> {
+    await super.initialize(dependencies, adapterSettings, endpointName, transportName)
+    this.provider = new ethers.providers.JsonRpcProvider(
+      adapterSettings.RPC_URL,
+      adapterSettings.CHAIN_ID,
+    )
+  }
+
+  async backgroundHandler(
+    context: EndpointContext<AddressTransportTypes>,
+    entries: RequestParams[],
+  ) {
+    if (!entries.length) {
+      await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
+      return
+    }
+    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
```

Comment:
Noticed from other migrated adapters that using backgroundExecute doesn't rate limit the adapter. So it triggers the next cycle as soon as one is done. To help mitigate this, let's sleep at the end of every background execute cycle
```suggestion
    await Promise.all(entries.map(async (param) => this.handleRequest(param)))
    await sleep(context.adapterSettings.BACKGROUND_EXECUTE_MS)
```